torch/_inductor/codegen/xpu/gemm_template.py

```diff
@@ -0,0 +1,922 @@
+# mypy: allow-untyped-defs
+import copy
+import enum
+import logging
+import re
+from abc import ABC, abstractmethod
+from typing import Optional, Union
+
+from ... import ir
+from ...config import sycl as inductor_sycl_config
+from ...ir import (
+    Buffer,
+    ChoiceCaller,
+    FixedLayout,
+    IRNode,
+    Layout,
+    SYCLTemplateBuffer,
+)
+from ...virtualized import V
+from ..common import IndentedBuffer
+from . import cutlass_utils
+from .sycl_kernel import SYCLTemplateKernel
+from .sycl_template import CUTLASSTemplate
+import torch
+
+log = logging.getLogger(__name__)
+
+# Jinja template for GEMM Kernel, used by the CUTLASSGemm3xTemplate class below.
+GEMM_TEMPLATE_CUTLASS_3X = r"""
+{{template.header().getvalue()}}
+{{template.globals().getvalue()}}
+{{instance_definition}}
+// When workspace_size is not a nullptr, populates requested workspace_size and returns.
+// Otherwise, computes the Gemm kernel using the given workspace ptr.
+extern "C" {
+PT_EXPORT {{kernel_call_signature}} {
+  try {
+  int B = {{kernel.size(Y, 0, -3, default_value=1)}};
+  using ElementComputeEpilogue = {{instance_type}}::ElementAccumulator;
+  using coord_t = cutlass::gemm::GemmCoord::Index;
+  static cutlass::KernelHardwareInfo hw_info;
+
+  const int device_id = 0;
```

Comment:
Do we expect the `device_id` to be always zero for XPU or is this just temporary for the PoC? If so, can we remove it or mark it with a `TODO(SYCL)`?